infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/limits/ram/ContainerResourceProvisioner.java

```diff
@@ -96,30 +92,39 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
 
         // make sure that machine configs have settings for RAM limit and request
         InternalMachineConfig machineConfig = machines.get(machineName(pod, container));
-        resourceLimitAttributesProvisioner.provisionMemory(
+        ResourceLimitAttributesProvisioner.provisionMemory(
             machineConfig,
             Containers.getRamLimit(container),
             Containers.getRamRequest(container),
             defaultMachineMaxMemorySizeAttribute,
             defaultMachineRequestMemorySizeAttribute);
 
         // make sure that machine configs have settings for CPU limit and request
-        resourceLimitAttributesProvisioner.provisionCPU(
+        ResourceLimitAttributesProvisioner.provisionCPU(
             machineConfig,
             Containers.getCpuLimit(container),
             Containers.getCpuRequest(container),
             defaultMachineCpuLimitAttribute,
             defaultMachineCpuRequestAttribute);
 
-        // reapply memory  and CPU settings to k8s container to make sure that provisioned
-        // values above are set
+        // reapply memory and CPU settings to k8s container to make sure that provisioned
+        // values above are set. Non-positive value means that limit is disabled, so just
+        // ignoring them.
         final Map<String, String> attributes = machineConfig.getAttributes();
-        Containers.addRamLimit(container, Long.parseLong(attributes.get(MEMORY_LIMIT_ATTRIBUTE)));
-        Containers.addRamRequest(
-            container, Long.parseLong(attributes.get(MEMORY_REQUEST_ATTRIBUTE)));
-        Containers.addCpuLimit(container, Float.parseFloat(attributes.get(CPU_LIMIT_ATTRIBUTE)));
-        Containers.addCpuRequest(
-            container, Float.parseFloat(attributes.get(CPU_REQUEST_ATTRIBUTE)));
+        long memLimit = Long.parseLong(attributes.get(MEMORY_LIMIT_ATTRIBUTE));
+        if (memLimit > 0) {
```

Comment:
what if I set `0` ?

Reply:
the same as -1

Reply:
can you please say it explicitly in che.properties? That values <= 0 are considered as not defined.

Reply:
ok